src/plugins/discover/public/application/main/components/layout/discover_histogram_content.tsx

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { EuiFlexGroup, EuiFlexItem, EuiHorizontalRule } from '@elastic/eui';
+import { SavedSearch } from '@kbn/saved-search-plugin/public';
+import React, { useCallback } from 'react';
+import { DataView } from '@kbn/data-views-plugin/common';
+import { METRIC_TYPE } from '@kbn/analytics';
+import { useDiscoverServices } from '../../../../hooks/use_discover_services';
+import { DataTableRecord } from '../../../../types';
+import { DocumentViewModeToggle, VIEW_MODE } from '../../../../components/view_mode_toggle';
+import { DocViewFilterFn } from '../../../../services/doc_views/doc_views_types';
+import { DataRefetch$, SavedSearchData } from '../../hooks/use_saved_search';
+import { AppState, GetStateReturn } from '../../services/discover_state';
+import { FieldStatisticsTable } from '../field_stats_table';
+import { DiscoverDocuments } from './discover_documents';
+import { DOCUMENTS_VIEW_CLICK, FIELD_STATISTICS_VIEW_CLICK } from '../field_stats_table/constants';
+
+const FieldStatisticsTableMemoized = React.memo(FieldStatisticsTable);
+
+export interface CommonDiscoverHistogramProps {
+  dataView: DataView;
+  savedSearch: SavedSearch;
+  isPlainRecord: boolean;
+  navigateTo: (url: string) => void;
+  savedSearchData$: SavedSearchData;
+  savedSearchRefetch$: DataRefetch$;
+  expandedDoc?: DataTableRecord;
+  setExpandedDoc: (doc?: DataTableRecord) => void;
+  viewMode: VIEW_MODE;
+  onAddFilter: DocViewFilterFn | undefined;
+  onFieldEdited: () => Promise<void>;
+  columns: string[];
+  state: AppState;
+  stateContainer: GetStateReturn;
+}
+
+export interface DiscoverHistogramContentProps extends CommonDiscoverHistogramProps {
+  chartHidden?: boolean;
+}
+
+export const DiscoverHistogramContent = ({
```

Comment:
I may rename this back to `DiscoverMainContent` since it doesn't contain the histogram. But `DiscoverHistogramLayout` for the other component seems good to me.